internal/status/gateway.go

```diff
@@ -40,24 +40,35 @@ func UpdateGatewayStatusProgrammedCondition(gw *gwapiv1b1.Gateway, svc *corev1.S
 			}
 		}
 
-		var gwAddrs []gwapiv1b1.GatewayAddress
+		if svc.Spec.ClusterIP != "" {
```

Comment:
shouldnt a check be added to make sure we are only doing this for svc type `ClusterIP` ? else we'll end up adding the `ClusterIP` for `LoadBalancer` type as well which we dont want

Reply:
done